include/device.h

```diff
@@ -725,18 +850,58 @@ static inline int device_pm_put_sync(const struct device *dev) { return -ENOTSUP
  * @}
  */
 
-#define Z_DEVICE_DEFINE(node_id, dev_name, drv_name, init_fn, pm_control_fn, \
-			data_ptr, cfg_ptr, level, prio, api_ptr)	\
+/** Synthesize the name of the object that holds device ordinal and
+ * dependency data.  If the object doesn't come from a devicetree
+ * node, use dev_name.
+ */
+#define Z_DEVICE_HANDLE_NAME(node_id, dev_name)				\
```

Comment:
if all of those macros are just going to be used in this file, I would just use _DEVICE_HANDLE_NAME etc.
This Z_ thing is currently very ambiguous and was introduced for internal private APIs across multiple files in the same modules, but IMO should be omitted for helper functions and  macros inside the same file.

Reply:
Those are *not* public API, they're an implementation detail of the public API that does use `DEVICE_HANDLE`.  They're globally visible and not to be used by anybody other than the implementation so I really think they should be namespaced.